pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -1154,3 +1180,441 @@ func (testVals *testGKENetworkParamSetController) doesGNPFinalizerExist(ctx cont
+					},
+					Labels: map[string]string{
+						labelsAddonManagerMode: ensureExistsMode,
+					},
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:           defaultTestNetworkName,
+					VPCSubnet:     defaultTestSubnetworkName,
+					PodIPv4Ranges: &networkv1.SecondaryRanges{RangeNames: []string{defaultPodRange}},
+				},
+			},
+			wantParamSet: desiredDefaultParamSet,
+		},
+		{
+			name: "revert label",
+			defaultParamSet: &networkv1.GKENetworkParamSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+					Annotations: map[string]string{
+						annotationComponentsLayer: componentLayer,
+						annotationComponentsName:  componentName,
+					},
+					Labels: map[string]string{
+						labelsAddonManagerMode: "",
+					},
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:           defaultTestNetworkName,
+					VPCSubnet:     defaultTestSubnetworkName,
+					PodIPv4Ranges: &networkv1.SecondaryRanges{RangeNames: []string{defaultPodRange}},
+				},
+			},
+			wantParamSet: desiredDefaultParamSet,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			g := gomega.NewGomegaWithT(t)
+			ctx, stop := context.WithCancel(context.Background())
+			defer stop()
+			testVals := setupGKENetworkParamSetController(ctx)
+
+			subnetKey := meta.RegionalKey(defaultTestSubnetworkName, testVals.clusterValues.Region)
+			subnet := &compute.Subnetwork{
+				Name: defaultTestSubnetworkName,
+				SecondaryIpRanges: []*compute.SubnetworkSecondaryRange{
+					{
+						IpCidrRange: defaultPodCIDR,
+						RangeName:   defaultPodRange,
+					},
+				},
+			}
+			err := testVals.cloud.Compute().Subnetworks().Insert(ctx, subnetKey, subnet)
+			if err != nil {
+				t.Error(err)
+			}
+
+			testVals.runGKENetworkParamSetController(ctx)
+
+			_, err = testVals.networkClient.NetworkingV1().Networks().Create(ctx, newL3Network(networkv1.DefaultPodNetworkName), metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create Network: %v", err)
+			}
+
+			_, err = testVals.networkClient.NetworkingV1().GKENetworkParamSets().Create(ctx, test.defaultParamSet, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create default GKENetworkParamSet: %v", err)
+			}
+
+			g.Eventually(func() (bool, error) {
+				paramSet, err := testVals.networkClient.NetworkingV1().GKENetworkParamSets().Get(ctx, networkv1.DefaultPodNetworkName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if paramSet.Spec.VPC != test.wantParamSet.Spec.VPC ||
+					paramSet.Spec.VPCSubnet != test.wantParamSet.Spec.VPCSubnet ||
+					!samePodIPv4Ranges(paramSet, test.wantParamSet) ||
+					!reflect.DeepEqual(paramSet.Annotations, test.wantParamSet.Annotations) ||
+					!reflect.DeepEqual(paramSet.ObjectMeta.Labels, test.wantParamSet.ObjectMeta.Labels) {
+					return false, fmt.Errorf("Default NetworkParamSet should be set to desired state")
```

Comment:
Would be good to print out the diff (want vs got).

Reply:
Done.